.github/workflows/goreleaser-snapshot-fleet.yaml

```
@@ -31,7 +31,7 @@ jobs:
     # Docker will not be available.
     #
     # Also do not run if author is dependabot or is on a fork (it doesn't have access to Github secrets).
-    if: ${{ (github.repository == 'fleetdm/fleet') && (github.actor != 'dependabot[bot]') && (github.event.pull_request.head.repo.full_name == github.repository)}}
+    if: ${{ (github.repository == 'fleetdm/fleet') && (github.actor != 'dependabot[bot]') && ((github.event_name == 'push') || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository)) }}
```

spitballing if this should be (I haven't tested it):

```suggestion
    if: ${{ (github.repository == 'fleetdm/fleet') && (github.actor != 'dependabot[bot]') && ((github.event_name == 'push' && github.ref_name == 'main') || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository)) }}
```